### ISSUE TITLE
yara-python install issue

### ISSUE THREAD

**Issue by nkleck** [plusvic/yara#258]
Ubuntu/Trusty64

Already installed yara binary - it works
installed python-dev

fails to build yara-python 
$ cd yara-python
$ python setup.py build

running build
running build_ext
building 'yara' extension
creating build
creating build/temp.linux-x86_64-2.7
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../libyara/include -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o
yara-python.c: In function ‘flo_read’:
yara-python.c:607:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < count; i++)
   ^
yara-python.c:607:3: note: use option -std=c99 or -std=gnu99 to compile your code
yara-python.c: In function ‘flo_write’:
yara-python.c:648:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < count; i++)
   ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

-creates the build dir, so tried:

$ sudo python setup.py install 
running install
running build
running build_ext
building 'yara' extension
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../libyara/include -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o
yara-python.c: In function ‘flo_read’:
yara-python.c:607:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < count; i++)
   ^
yara-python.c:607:3: note: use option -std=c99 or -std=gnu99 to compile your code
yara-python.c: In function ‘flo_write’:
yara-python.c:648:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < count; i++)
   ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
also fails if try pip install:

$ pip install yara-python/
Unpacking ./yara-python
  Running setup.py (path:/tmp/pip-qcT6iO-build/setup.py) egg_info for package from file:///vagrant/dev/yara/yara-python
    
Installing collected packages: yara-python
  Running setup.py install for yara-python
    building 'yara' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../libyara/include -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o
    yara-python.c: In function ‘flo_read’:
    yara-python.c:607:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
       for (int i = 0; i < count; i++)
       ^
    yara-python.c:607:3: note: use option -std=c99 or -std=gnu99 to compile your code
    yara-python.c: In function ‘flo_write’:
    yara-python.c:648:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
       for (int i = 0; i < count; i++)
       ^
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-qcT6iO-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-nsrSwy-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_ext

building 'yara' extension

creating build

creating build/temp.linux-x86_64-2.7

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../libyara/include -I/usr/include/python2.7 -c yara-python.c -o build/temp.linux-x86_64-2.7/yara-python.o

yara-python.c: In function ‘flo_read’:

yara-python.c:607:3: error: ‘for’ loop initial declarations are only allowed in C99 mode

   for (int i = 0; i < count; i++)

   ^

yara-python.c:607:3: note: use option -std=c99 or -std=gnu99 to compile your code

yara-python.c: In function ‘flo_write’:

yara-python.c:648:3: error: ‘for’ loop initial declarations are only allowed in C99 mode

   for (int i = 0; i < count; i++)

   ^

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-qcT6iO-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-nsrSwy-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip-qcT6iO-build
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==1.5.4', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 185, in main
    return command.main(cmd_args)
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 31: ordinal not in range(128)

**Comment by plusvic** [comment]
I used to declare loop variables out of the "for" statement to comply with ANSI C, but nowadays most compilers (if not all) are supporting C99, so I prefer to use the more convenient C99 style. In fact I've been replacing the old ANSI C style with C99 style whenever I touch old code. I guess we can pass the 
-std=c99 argument somwhere in setup.py

**Comment by plusvic** [comment]
Fixed in 4994a2cdff17e54b5e55507e494223abdca441dd
